omod/src/main/webapp/fragments/htmlform/enterHtmlForm.gsp

```diff
@@ -65,20 +65,27 @@
 
         <% if (visit) { %>
             <% if (command.context.mode.toString().equals('ENTER') && !visit.isOpen()) { %>
-                // set default date to the visit start date for retrospective visits
-                htmlForm.setEncounterDate(moment('${ ui.dateToISOString(visit.startDate).split('T')[0] }').toDate());
+                <% if (ui.convertTimezones()) { %>
+                    //New encounter for past visit
+                    htmlForm.adjustTimeZoneEncounterDate('${visitStartDatetime}');
+                <% } else {%>
+                    // set default date to the visit start date for retrospective visits
+                    htmlForm.setEncounterDate('${ visitStartDatetime}');
+                <% } %>
             <% } %>
-
-            // set valid date range based on visit
-            htmlForm.setEncounterStartDateRange(moment('${  ui.dateToISOString(visit.startDate).split('T')[0] }').toDate());
-            htmlForm.setEncounterStopDateRange(moment('${ visit.stopDate ? ui.dateToISOString(visit.stopDate).split('T')[0] : ui.dateToISOString(currentDate).split('T')[0] }').toDate());
-
+            // set valid date range based on visit getStartDatetime
+            htmlForm.setEncounterStartDateRange('${visitStartDatetime}', ${ui.convertTimezones()});
+            htmlForm.setEncounterStopDateRange('${visitStopDatetime}', ${ui.convertTimezones()});
         <% } else { %>
             // note that we need to get the current datetime from the *server*, in case the server and client are in different time zones
-            htmlForm.setEncounterStopDateRange(moment('${  ui.dateToISOString(currentDate).split('T')[0] }').toDate());
-            htmlForm.setEncounterDate(moment('${  ui.dateToISOString(currentDate).split('T')[0] }').toDate());
+            htmlForm.setEncounterDate('${(currentDate)}');
+            //set client Timezone
+            <% if (ui.convertTimezones() != null) { %>
```

Comment:
Generally speaking, the only changes to existing code that i would expect are those where convertTimeZones is true.

Reply:
The difference here is that before we strip the date from time on server side, now we do that on the client side, here :
https://github.com/openmrs/openmrs-module-htmlformentryui/pull/40/files/64a1b28d878640245daa824ad0c0c469f1e5d232#diff-53c10c5591297c3a544b761c5e9856c108393e5d68c715daabf8b4810697b03eR59

But maybe we can go back with this and create 2 completely different behaviors, one like it was before (split date from time on server side) and the other for timezones (on client side). CC @dkayiwa